app/models/shipit/stack.rb

```diff
@@ -116,6 +117,18 @@ def head
       commits.reachable.first.try!(:sha)
     end
 
+    def merge_status
+      if locked?
+        'locked'
+      else
+        significant_commits = commits.reachable.where('id >= ?', last_deployed_commit).order(id: :desc)
```

Comment:
I'm not sure the last deployed commit actually matter. `id > ?` is what I'd use. So `...reacheable.newer_than(last_deployed_commit)...`